### PR TITLE
Fix unstack

### PR DESCRIPTION
**Pull request by juliettelavoie**
### Pull Request Checklist:
- [x] `pre-commit` hooks are installed/active in my local clone (`$ pre-commit install`)
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [ ] If a merge request has been made in parallel to this PR in xscen-notebooks, it is merged and the submodules have been updated.
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* ``xs.utils.stack_drop_nans``/ ``xs.utils.unstack_fill_nan``  format the `to_file`/`coords` string to add the domain and the initial shape. 
* The initial shape is added to the lat and lon attrs to be able to retrive it when it is time to unstack.
* Redifining lon and lat to fix bug in xarray (https://github.com/pydata/xarray/issues/6946)

### Does this PR introduce a breaking change?
I don't think so. If you pass a string without {domain} or {shape} as before, the formatting will be ignored.

### CODE REVIEW

File: xscen/utils.py [hunk 1]
```diff
@@ -106,13 +116,39 @@ def stack_drop_nans(
...
+        .where(mask_1d, drop=True)
+        .reset_index(new_dim, drop=True)
+    )
     for dim in mask.dims:
         out[dim].attrs.update(ds[dim].attrs)
 
     if to_file is not None:
+        # set default path to store the information necessary to unstack
+        # the name includes the domain and the original shape to uniquely identify the dataset
+        domain = ds.attrs.get("cat:domain", "unknown_domain")
```
Comment (aulemahal) [inline xscen/utils.py:134]: Je suggère pas de '_'
```suggestion
        domain = ds.attrs.get("cat:domain", "unknown")
```

File: xscen/utils.py [hunk 2]
```diff
@@ -106,13 +116,39 @@ def stack_drop_nans(
     --------
     unstack_fill_nan : The inverse operation.
     """
+
+    original_shape = f"{len(ds.lon)}x{len(ds.lat)}"
```
Comment (aulemahal) [inline xscen/utils.py:120]: Je sais pas à quel point c'était utile, mais l'ancienne version de cette fonction était indépendante des noms de dimensions. On utilise `mask.dims`.
Est-ce que ça serait à garder?
Si oui, je propose:
```suggestion
    original_shape = "x".join(map(str, mask.shape))
```
et il faudrait rendre le reste du code générique aux dims de `mask`...

Et si non, il faudrait juste le dire dans la fonction qu'on s'attend à `lat` et `lon`.

**Review by aulemahal — COMMENTED** [review]
Ça me semble une bonne solution!
Petit questionnement sur la généricité (?) à conserver?

File: xscen/utils.py (hunk 2 above)
Comment (juliettelavoie) [reply xscen/utils.py:120]: ah ben oui ça serait bien de garder cette flexibilité là. Je vais faire les changements.

Comment (juliettelavoie) [reply xscen/utils.py]: The changes I have added will fail if one of the dimensions of ds_stack have an unrelated attrs also named `original_shape`, but that feels very unlikely...

**Review by aulemahal — APPROVED** [review]

**Pull request merged**